hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -896,8 +900,15 @@ def hardy_weinberg_test(expr) -> StructExpression:
     :class:`.StructExpression`
         Struct expression with fields `het_freq_hwe` and `p_value`.
     """
-    t = tstruct(het_freq_hwe=tfloat64, p_value=tfloat64)
-    return _agg_func('HardyWeinberg', [expr], t)
+    return hl.rbind(
+        hl.rbind(
+            expr,
+            lambda call: hl.agg.filter(call.ploidy == 2, hl.agg.counter(call.n_alt_alleles())
+                                       .map_values(lambda i: hl.case()
+                                                   .when(i < 1 << 31, hl.int(i))
+                                                   .or_error('hardy_weinberg_test: count greater than MAX_INT'))),
+            _ctx='scan' if getattr(hardy_weinberg_test, '_as_scan', False) else 'agg'),
```

Comment:
this somehow feels like it's flipped? (can we abstract it away under an `hl.agg._bind`?)

Reply:
(also I think this should be `_agg_func._as_scan is False`)